webapp/shop/cred/views.py

```diff
@@ -171,12 +174,14 @@ def cred_sign_up(**_):
     is_staging = "staging" in os.getenv(
         "CONTRACTS_API_URL", "https://contracts.staging.canonical.com/"
     )
-    form_fields["enviornment"] = "staging" if is_staging else "production"
+    marketo_form_id = "6254" if is_staging else "3801"
     payload = {
-        "formId": form_fields.pop("formid"),
+        "formId": marketo_form_id,
         "input": [
             {
-                "leadFormFields": form_fields,
+                "leadFormFields": form_fields.update(
```

Comment:
dict.update is a void function and does the update in place
Please do before line 179:
`form_fields.update({"formid": marketo_form_id})`

And then the current line 179 will be unchanged